### DISCUSSION TITLE
radio buttons are not selected by default when using with useController in NextJs app but working in react app

### DISCUSSION THREAD

**Discussion by rawatnaresh** [react-hook-form/react-hook-form#11563] [Q&A] [answered]
I've simple nextjs page in which I'm using `useController` with radio buttons which have some default values but the radio button is not selected based on default values. 

Radios are only selected by default when using with `register`
[Here's codesandbox example](https://codesandbox.io/p/sandbox/next-react-hook-form-5h78js)

### NextJS example with react-hook-form

```tsx
"use client";

import { useController, Control, useForm } from "react-hook-form";

function TypeSelector({ control }: { control: Control<any> }) {
  const { field } = useController({
    control,
    name: "type",
    defaultValue: "text",
  });

  return (
    <div>
      <p>useController::Default type should be text</p>

      <label>
        <input type="radio" {...field} value={"media"} />
        Media
      </label>
      <label>
        <input type="radio" {...field} value={"text"} />
        Text
      </label>
    </div>
  );
}
export default function Home() {
  const { control, register } = useForm({
    defaultValues: {
      type: "text",
      gender: "male",
    },
  });
  return (
    <div>
      <TypeSelector control={control} />
      <p>register::Default gender should be male</p>
      <label>
        <input type="radio" {...register("gender")} value={"male"} />
        Male
      </label>
      <label>
        <input type="radio" {...register("gender")} value={"female"} />
        Female
      </label>
    </div>
  );
}

```
<img width="715" alt="Screenshot 2024-02-28 at 23 06 14" src="https://github.com/react-hook-form/react-hook-form/assets/29520305/477c180d-8b6c-4cd2-94ce-59cf0677c8c9">

But same thing is working fine in react.

[Codesandbox exmaple in react](https://codesandbox.io/p/sandbox/react-react-hook-form-6j2fd4)

### ReactJS example with react-hook-form

```tsx
import { useController, Control, useForm } from "react-hook-form";

function TypeSelector({ control }: { control: Control<any> }) {
  const { field } = useController({
    control,
    name: "type",
    defaultValue: "text",
  });

  return (
    <div>
      <p>useController::Default type should be text</p>

      <label>
        <input type="radio" {...field} value={"media"} />
        Media
      </label>
      <label>
        <input type="radio" {...field} value={"text"} />
        Text
      </label>
    </div>
  );
}
export default function Home() {
  const { control, register } = useForm({
    defaultValues: {
      type: "text",
      gender: "male",
    },
  });
  return (
    <div>
      <TypeSelector control={control} />
      <p>register::Default gender should be male</p>
      <label>
        <input type="radio" {...register("gender")} value={"male"} />
        Male
      </label>
      <label>
        <input type="radio" {...register("gender")} value={"female"} />
        Female
      </label>
    </div>
  );
}

```
<img width="716" alt="Screenshot 2024-02-28 at 23 17 10" src="https://github.com/react-hook-form/react-hook-form/assets/29520305/d17ceb9a-4335-4c56-b580-cbc04ba414ba">

**Comment by sujinleeme** [comment, marked as answer]
Hi, 

1. React hook form version

First of all, the **ReactJS example with react-hook-form** uses "react-hook-form": "7.4.1" whereas the next js version uses "react-hook-form": "7.50".  If you update to the latest version,  you can observe the same behaviour; the first radio button doesn't set the default value.

Probably, it looks like the old react-hook-form's controller updates the `checked` like `register()` by refs.

![image](https://github.com/react-hook-form/react-hook-form/assets/6011401/c1130bd1-6898-4cc9-87a8-346d969c3737)


2. How to select the default value on the radio input with useController 

You can simply add the `defaultChecked` or `checked` prop, but not both.

```ts
<input type="radio" {...field} value={"media"} defaultChecked />
```

Also, you don't need to set defaultValue again because you already give defaultValues in the useForm.

```ts
  const { field } = useController({
    control,
    name: "type",
  });
```


Here is the next.js version [codesandbox](https://codesandbox.io/p/devbox/11563-lhj42x?file=%2Fapp%2Fpage.tsx%3A36%2C7&workspaceId=eb324436-cc8e-4656-aa7d-4d9bdfcc1c39).
